pypesto/optimize/util.py

```diff
@@ -0,0 +1,68 @@
+from ..engine import Engine, SingleCoreEngine
+from ..objective import HistoryOptions
+from ..store.save_to_hdf5 import get_or_create_group
+from pathlib import Path
+
+
+import h5py
+
+
+def check_hdf5_mp(history_options: HistoryOptions,
+                  engine: Engine):
+    """
+    Create a folder for partial HDF5 files,
+    if a parallelization engine will be used.
+
+    Paramters
+    ---------
+    engine:
+        The Engine which is used in the optimization
+    history_options:
+        The HistoryOptions used in the optimization
+
+    Returns
+    -------
+    The filename that will be used to combine the partial HDF5 files later.
+    If a parallelization engine is not used, `None` is returned.
+    """
+    if not isinstance(engine, SingleCoreEngine):
```

Comment:
Why don't you go for 
```python
if isinstance(engine, SingleCoreEngine):
   return None
filename = ...
```
Would be more intuitive to read for future contributors... :)